dfs/nfs/server.go

```diff
@@ -398,6 +398,40 @@ func (c *Server) cleanupBindMounts() {
 			}
 		}
 	}
+	// CC-1816: clean up remnants of old /exports/serviced_var_volumes directory
+	removeDeprecated("/exports/serviced_var_volumes")
+}
+
+// removeDeprecated will clean up any old exports path
+func removeDeprecated(dirpath string) {
+	if dirContents, err := ioutil.ReadDir(dirpath); !os.IsNotExist(err) {
+		if err != nil {
+			glog.Warningf("Could not look up deprecated exports path %s: %s", dirpath, err)
+			return
+		}
+		// Unmount path
+		if mounted, err := isBindMounted(dirpath); err != nil {
+			glog.Warningf("Could not determine if deprecated path %s is bind mounted: %s", dirpath, err)
+		} else if mounted {
+			if err := umount(dirpath); err != nil {
+				glog.Warningf("Could not unmount deprecated path %s: %s", dirpath, err)
+			} else {
+				glog.Infof("Unmounted deprecated path %s", dirpath)
+			}
+		}
+		// recurse through the filesystem
+		for _, file := range dirContents {
+			if file.IsDir() {
+				removeDeprecated(filepath.Join(dirpath, file.Name()))
+			}
+		}
+		// remove the path
+		if err := os.RemoveAll(dirpath); err != nil {
```

Comment:
what this is doing is 
1) unmounting the base path
2) unmount mounted children
3) delete mounted children
4) delete basepath